File: python/gherkin3/gherkin_line.py
class GherkinLine(object):
    def __init__(self, line_text, line_number):
        self._line_text = line_text
        self._line_number = line_number
        self._trimmed_line_text = line_text.lstrip()
        self.indent = len(line_text) - len(self._trimmed_line_text)

    def get_rest_trimmed(self, length):
        return self._trimmed_line_text[length:].strip()

    def get_line_text(self, indent_to_remove=-1):
        if indent_to_remove < 0 or indent_to_remove > self.indent:
            return self._trimmed_line_text
        else:
            return self._line_text[indent_to_remove:]

    def is_empty(self):
        return not self._trimmed_line_text

    def startswith(self, prefix):
        return self._trimmed_line_text.startswith(prefix)

    def startswith_title_keyword(self, keyword):
        return self._trimmed_line_text.startswith(keyword + ':')

    def table_cells(self):
        column = self.indent + 1
        items = self._trimmed_line_text.strip().split('|')
        cells = []
        for item in items[1:-1]:
            cell_indent = len(item) - len(item.lstrip()) + 1
            cells.append({'column': column + cell_indent, 'text': item.strip()})
            column += len(item) + 1
        return cells

    def tags(self):
        column = self.indent + 1
        items = self._trimmed_line_text.strip().split('@')
        tags = []
        for item in items[1:]:
            tags.append({'column': column, 'text': '@' + item.strip()})
            column += len(item) + 1
        return tags

File: swift/xcode_proj/gherkin/gherkin/GherkinLine.swift
import Foundation

class GherkinLine {
    let lineText: String
    var lineNumber: Int
    var trimmedLineText: String
    var indent: Int {
        return lineText.characters.count - trimmedLineText.characters.count
    }
    
    init(lineText: String, lineNumber: Int) {
        
        self.lineText = lineText
        self.lineNumber = lineNumber
        self.trimmedLineText = self.lineText.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet())
    }
    
    func startsWith(prefix: String) -> Bool {
        
        return self.trimmedLineText.hasPrefix(prefix)
    }
    
    func startsWithTitleKeyword(keyword: String) -> Bool {
        
        return self.startsWith("\(keyword):")
    }
    
    /**
        give a trimmed version of trimmedLineText
    
        :param: length Int to trim from front of text
    
        :return: String
    */
    func getRestTrimmed(length: Int) -> String {
        
        let range = Range(start: trimmedLineText.startIndex.advancedBy(length), end: trimmedLineText.endIndex)

        let trimmed = trimmedLineText.substringWithRange(range)
        return trimmed
    }
    
    /**
        check if trimmedLineText is empty

        :return: Bool
    */
    func empty() -> Bool {
        
        return trimmedLineText.isEmpty
    }
    
    /**
        Get the text with removed amount
    
        :param: indexToRemove Int
    
        :return: String
    */
    func getLineText(indexToRemove: Int = 0) -> String {
        
        if indexToRemove < 0 || indexToRemove > indent {
            return trimmedLineText
        } else {
            let range = Range(start: lineText.startIndex.advancedBy(indexToRemove), end: lineText.endIndex)
            let trimmed = lineText.substringWithRange(range)
            return trimmed
        }
    }
    
    func tableCells() -> Array<Dictionary<Int, String>> {
        let line = trimmedLineText
        let columns = line.componentsSeparatedByString("|").map {
            $0.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet())
            }.filter {$0 != ""}
        
        var indexedColumns = [[Int: String]]()
        for (index, value) in columns.enumerate() {
            indexedColumns.append([index: value])
        }
        return indexedColumns
    }
    
    func tags() -> Dictionary<Int, String> {
        
        var column = indent + 1
        var items = trimmedLineText.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet()).componentsSeparatedByString("@")
    }
}

















